src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -977,6 +978,9 @@ def build_proto_from_callable(
         # Use the actual function name. Note: We do not support different scopes yet, need to be careful with this then.
         expr_builder.name = func.__name__
 
+    if object_name is not None:
+        build_sp_table_name(expr_builder.object_name, object_name)
+
 
```

Comment:
should we update udaf/sproc as well?

Reply:
Yes, I was planning to do in another pr after this one merges.